File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for(let [i, el] of nums.entries()){
        for(let j=i+1; j<nums.length;j++){
            if(target === nums[j] + el){
                return [i, j]
            }
        }
    }
    return [];
};

File: 0001-two-sum/0001-two-sum.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> num_indices;

        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];

            if (num_indices.find(complement) != num_indices.end()) {
                return {num_indices[complement], i};
            }

            num_indices[nums[i]] = i;
        }

        // If no solution is found, return an empty vector or handle it as appropriate
        return {};
    }
};
